### PR TITLE
feat: rework cache & add gcs cache option

### PR DESCRIPTION
**Pull request by matthisholleville**
Closes # 

## 📑 Description

Rework cache & add GCS cache option to match with this PR https://github.com/k8sgpt-ai/k8sgpt/pull/750

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

## ℹ Additional Information

PR dependencies :
- Core: https://github.com/k8sgpt-ai/k8sgpt/pull/750
- Schemas: https://github.com/k8sgpt-ai/schemas/pull/19

### CODE REVIEW

**Review by AlexsJones — APPROVED** [review]

**Review by arbreezy — DISMISSED** [review]
lgtm , lets merge after the schema PR is merged too

File: config/samples/core_v1alpha1_k8sgpt.yaml
Comment (AlexsJones) [inline config/samples/core_v1alpha1_k8sgpt.yaml:19]: maybe have an example for each?

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**
